legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabWorkspaceApi.java

```diff
@@ -585,7 +585,7 @@ public WorkspaceUpdateReport updateWorkspace(String projectId, String workspaceI
             String workspaceCreationRevisionId;
             try
             {
-                workspaceCreationRevisionId = withRetries(() -> repositoryApi.getMergeBase(gitLabProjectId.getGitLabId(), Arrays.asList(MASTER_BRANCH, currentWorkspaceRevisionId)).getId());
+                workspaceCreationRevisionId = withRetries(() -> repositoryApi.getMergeBase(gitLabProjectId.getGitLabId(), Arrays.asList(getDefaultBranch(gitLabProjectId), currentWorkspaceRevisionId)).getId());
```

Comment:
Try to avoid calling getDefaultBranch multiple times. You're already calling it on line 510.